From cd930a17782b02d4d6743f5d4ea05379304fe140 Mon Sep 17 00:00:00 2001
From: ejaaskel <esa.jaaskela@suomi24.fi>
Date: Fri, 5 Dec 2025 08:51:30 +0000
Subject: [PATCH] initscripts: Fix date calls

Fix the bootmisc and save-rtc date calls so that they work as expected
with uutils-coreutils.

The first issue is with the format width modifiers. These have been
corrected in the newer versions of the coreutils (tested on 0.4.0).

The second issue is with the time set accepting only a certain date
format, and the requirement for -s flag when setting date. This still
seems to be present in the uutils-coreutils master.

Upstream-Status: Inappropriate [Sulka specific]
---
 .../initscripts/initscripts-1.0/bootmisc.sh            | 10 +++-------
 1 file changed, 3 insertions(+), 7 deletions(-)

diff --git a/meta/recipes-core/initscripts/initscripts-1.0/bootmisc.sh b/meta/recipes-core/initscripts/initscripts-1.0/bootmisc.sh
index 6068a8c07f..b866223fc0 100755
--- a/bootmisc.sh
+++ b/bootmisc.sh
@@ -73,15 +73,11 @@ fi
 test -x /etc/init.d/hwclock.sh && /etc/init.d/hwclock.sh start
 if test -e "$TIMESTAMP_FILE"
 then
-	SYSTEMDATE=`date -u +%4Y%2m%2d%2H%2M%2S`
+	SYSTEMDATE=`date -u +%Y%m%d%H%M%S`
 	read TIMESTAMP < "$TIMESTAMP_FILE"
 	if [ ${TIMESTAMP} -gt $SYSTEMDATE ]; then
-		# format the timestamp as date expects it (2m2d2H2M4Y.2S)
-		TS_YR=${TIMESTAMP%??????????}
-		TS_SEC=${TIMESTAMP#????????????}
-		TS_FIRST12=${TIMESTAMP%??}
-		TS_MIDDLE8=${TS_FIRST12#????}
-		date -u ${TS_MIDDLE8}${TS_YR}.${TS_SEC}
+		# Format the timestamp as YYYY-mm-dd HH:MM:SSZ
+		date -s "$(cat /etc/timestamp | sed 's/\(....\)\(..\)\(..\)\(..\)\(..\)\(..\)/\1-\2-\3 \4:\5:\6Z/')"
 		test -x /etc/init.d/hwclock.sh && /etc/init.d/hwclock.sh stop
 	fi
 fi
--- a/save-rtc.sh
+++ b/save-rtc.sh
@@ -18,4 +18,4 @@ TIMESTAMP_FILE=/etc/timestamp
 [ -f /etc/default/timestamp ] && . /etc/default/timestamp
 
 # Update the timestamp
-date -u +%4Y%2m%2d%2H%2M%2S 2>/dev/null > "$TIMESTAMP_FILE"
+date -u +%Y%m%d%H%M%S 2>/dev/null > "$TIMESTAMP_FILE"
-- 
2.34.1

